shopinvader_api_delivery_carrier/routers/delivery.py

```diff
@@ -0,0 +1,70 @@
+# Copyright 2023 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from typing import Annotated
+
+from fastapi import APIRouter, Depends
+
+from odoo import api, fields, models
+
+from odoo.addons.base.models.res_partner import Partner as ResPartner
+from odoo.addons.extendable_fastapi.schemas import PagedCollection
+from odoo.addons.fastapi.dependencies import (
+    authenticated_partner,
+    authenticated_partner_env,
+    paging,
+)
+from odoo.addons.fastapi.schemas import Paging
+from odoo.addons.shopinvader_filtered_model.utils import FilteredModelAdapter
+from odoo.addons.stock.models.stock_picking import Picking as StockPicking
+
+from ..schemas import Picking
+
+delivery_router = APIRouter(tags=["deliveries"])
+
+
+@delivery_router.get("/deliveries")
+def search(
+    env: Annotated[api.Environment, Depends(authenticated_partner_env)],
+    partner: Annotated["ResPartner", Depends(authenticated_partner)],
+    paging_: Annotated[Paging, Depends(paging)],
+) -> PagedCollection[Picking]:
```

Comment:
It seems that in the version 14, it was possible for an anonymous user to search on deliveries, but he has always to get an empty recordset.
I'm not sure of the way of implementing it here, and not sure it is needed (since he won't get any info).

Reply:
The route is authenticated now, we don't need to deal with anonymous user.